cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -484,17 +501,70 @@ class model : public base {
         return get_class_count_impl();
     }
 
+    /// Performs Depth First Traversal of i-th tree
+    /// @param[in] tree_idx     Index of the tree to traverse
+    /// @param[in] visitor      This functor gets notified when tree nodes are visited, via corresponding operators:
+    ///                             bool operator()(const decision_forest::split_node_info<Task>&)
+    ///                             bool operator()(const decision_forest::leaf_node_info<Task>&)
+    template <typename Visitor>
+    void traverse_dfs(std::int64_t tree_idx, Visitor&& visitor) const {
+        traverse_dfs_impl(tree_idx,
+                          dt::detail::make_node_visitor<dtree_task_t>(std::move(visitor)));
+    }
+
+    /// Performs Breadth First Traversal of i-th tree
+    /// @param[in] tree_idx    Index of the tree to traverse
+    /// @param[in] visitor      This functor gets notified when tree nodes are visited, via corresponding operators:
+    ///                             bool operator()(const decision_forest::split_node_info<Task>&)
+    ///                             bool operator()(const decision_forest::leaf_node_info<Task>&)
+    template <typename Visitor>
+    void traverse_bfs(std::int64_t tree_idx, Visitor&& visitor) const {
+        traverse_bfs_impl(tree_idx,
+                          dt::detail::make_node_visitor<dtree_task_t>(std::move(visitor)));
+    }
+
 protected:
     std::int64_t get_class_count_impl() const;
 
+    void traverse_dfs_impl(std::int64_t tree_idx, visitor_t&& visitor) const;
+    void traverse_bfs_impl(std::int64_t tree_idx, visitor_t&& visitor) const;
+
 private:
     explicit model(const std::shared_ptr<detail::model_impl<Task>>& impl);
     dal::detail::pimpl<detail::model_impl<Task>> impl_;
 };
 
+template <typename Task>
+using node_info = dt::node_info<typename decision_tree_task_map<Task>::tree_task_t>;
```

Comment:
Suggest to introduce `using decision_tree_task_map_t = typename decision_tree_task_map<Task>::tree_task_t`, so it can be rewritten:
```
template <typename Task>
using node_info = dt::node_info<decision_tree_task_map_t<Task>::tree_task_t>;
```

Reply:
applied